package ru.kt15.net.labs.sessions;

message SessionKey {
	required string sessionId = 1;
	required string serverId = 2;
}

message Session {
	required SessionKey key = 1;
	required uint64 timestamp = 2;
	optional string sessionSource = 3;
	optional string sessionDest = 4;
	optional uint64 validUntil = 5;
	optional bool remove = 6;
}

message HostKey {
	required string address = 1;
}

message Host {
	required HostKey key = 1;
	required uint64 timestamp = 2;
	optional string name = 3;
	optional bool valid = 4;
	optional bool remove = 5;
}

message List {
	repeated Session sessions = 1;
	repeated Host hosts = 2;
}
